xarray/coding/cftime_offsets.py

```diff
@@ -171,6 +172,47 @@ def _get_offset_day(self, other):
         return _get_day_of_month(other, self._day_option)
 
 
+class Tick(BaseCFTimeOffset):
+    def _next_higher_resolution(self):
+        if type(self) is Day:
+            return Hour(self.n * 24)
+        if type(self) is Hour:
+            return Minute(self.n * 60)
+        if type(self) is Minute:
+            return Second(self.n * 60)
+        if type(self) is Second:
+            return Millisecond(self.n * 1000)
+        if type(self) is Millisecond:
+            return Microsecond(self.n * 1000)
+        # if type(self) is Microsecond:
+        #    return Nanosecond(self.n * 1000)
+        raise TypeError(
+            "The provided multiple 'n' must be an integer or float. "
+            "Instead a value of type {!r} was provided.".format(type(self.n))
+        )
+
+    def __init__(self, n: int | float = 1.):
+        if not isinstance(n, (int, float)):
+            raise TypeError(
+                "The provided multiple 'n' must be an integer or float. "
+                "Instead a value of type {!r} was provided.".format(type(n))
+            )
+        self.n = n
```

Comment:
dont understand the `mypy` error here: https://results.pre-commit.ci/run/github/13221727/1644324423.gojmV_7_TByQ6sxITgwjHw

Reply:
```
class BaseCFTimeOffset:
    n: int | float
    def __init__(n: int)

class Tick(BaseCFTimeOffset):
    n: int | float